.github/workflows/release.yml

```diff
@@ -0,0 +1,241 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v*' # Trigger on any tag starting with 'v' (e.g., v0.1.0, v1.0.0-beta)
+  workflow_dispatch: # Allow manual trigger for testing
+
+permissions:
+  contents: write # Required for creating releases
+
+jobs:
+  release:
+    name: Build and Publish Module
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Fetch all history for changelog
+
+      - name: Setup PowerShell
+        shell: pwsh
+        run: |
+          Write-Host "PowerShell Version: $($PSVersionTable.PSVersion)" -ForegroundColor Cyan
+          Write-Host "OS: $($PSVersionTable.OS)" -ForegroundColor Cyan
+
+      - name: Extract version from tag
+        id: version
+        shell: bash
```

Comment:
change this to powershell please and the script body in this job should use powershell